sys/arm64/arm64/locore.S

```diff
@@ -890,10 +890,26 @@ initstack:
 	.space	(PAGE_SIZE * KSTACK_PAGES)
 initstack_end:
 
+/*
+ * sigcode has to be labeled as an #object type so that the symbols
+ * resolve to the correct address as a source for copies.  This also
+ * ensures that captable pointers to it will be able to read it.  This
+ * is fine as the code is never executed directly in the kernel, just
+ * copied to places for userland to execute.
+ */
+#define	SIGCODE(sym)						\
+	.text; .globl sym; .align 2; .type sym,#object; sym:
```

Comment:
I guess this means `.align` is power of two for the AArch64 toolchain? Maybe use .p2align/.balign to be explicit?

Reply:
This is copied from `ENTRY()` from sys/arm64/include/asm.h.  I don't have a strong opinion on which one is used, but this should be in sync with whatever `ENTRY()` uses, just s/function/object/.

Reply:
Maybe put it in asm.h? It'll be needed for linux_locore.

Reply:
Hmm, for the VDSO?  In that case, it might need a better name than `SIGCODE`.  `CODEASDATA` is yuck, `ENTRY_DATA` is slightly less terrible.  `ENTRY_RO` might be another candidate.

Reply:
I'm inclined to maybe move it to asm.h as a followup once we need it for the VDSO, mostly to simplify the next merge to morello-dev.

Reply:
Makes sense, can do other architectures at the same time too.